src/core/qgsiconutils.cpp

```diff
@@ -164,6 +165,7 @@ QIcon QgsIconUtils::iconForLayerType( QgsMapLayerType type )
       return QgsApplication::getThemeIcon( QStringLiteral( "/mIconAnnotationLayer.svg" ) );
 
     case QgsMapLayerType::PluginLayer:
+    case QgsMapLayerType::GroupLayer:
```

Comment:
You could simply use mActionFolder.svg (if the plan is to get rid of the old 'groups'). Otherwise, we can come up with a derived version based on that icon later.

Reply:
This won't be visible anywhere at all -- group layers will always be a hidden concept from users.